server/src/main/java/app/friendship/FriendshipController.java

```diff
@@ -18,60 +19,70 @@ public class FriendshipController {
 
     @PostMapping("/sendRequest")
     @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    @PreAuthorize("@jwtService.hasAccess(authentication, #userId)")
     public void sendFriendRequest(@PathVariable Integer userId, @RequestBody FriendshipRequest request) {
         friendshipService.sendFriendRequest(userId, request.receiverId());
     }
 
     @PostMapping("/accept")
     @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    @PreAuthorize("@jwtService.hasAccess(authentication, #userId)")
     public void acceptFriendRequest(@PathVariable Integer userId, @RequestBody FriendshipRequest request) {
         friendshipService.acceptFriendRequest(userId, request.receiverId());
     }
 
     @PostMapping("/remove")
     @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    @PreAuthorize("@jwtService.hasAccess(authentication, #userId)")
     public void removeFriend(@PathVariable Integer userId, @RequestBody FriendshipRequest request) {
         friendshipService.removeFriend(userId, request.receiverId());
     }
 
     @GetMapping("/request")
     @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    @PreAuthorize("@jwtService.hasAccess(authentication, #userId)")
     public List<Integer> getFriendRequests(@PathVariable Integer userId) {
         return friendshipService.getPendingFriendRequests(userId);
     }
 
     @GetMapping("/sent")
     @Operation(security = {@SecurityRequirement(name = "JwtAuth")})
+    @PreAuthorize("@jwtService.hasAccess(authentication, #userId)")
     public List<Integer> getSentFriendRequests(@PathVariable Integer userId) {

```

Comment:
da sie to polepszyc zaraz pwiem jak. 

https://chat.deepseek.com/a/chat/s/be1b659b-e43d-4f79-8663-7ebd875a8de1